src/site/content/en/blog/optimize-long-tasks/index.md

```diff
@@ -0,0 +1,403 @@
+    const task = tasks.shift();
+
+    // Run the task:
+    task();
+
+    // Yield to the main thread:
+    await yieldToMain();
+  }
+}
+```
+
+{% Aside 'important' %}
+You don't have to yield after _every_ function call. For example, if you run two functions that result in critical updates to the user interface, you probably don't want to yield in between them. If you can, let that work run _first_, then consider yielding in between functions that do less critical or background work that the user doesn't see.
+{% endAside %}
+
+The result is that the once-monolithic task is now broken up into separate tasks.
+
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/wg0FW6S29CzOCbbwk9kK.png", alt="The same saveSettings function depicted in Chrome's performance profiler, only with yielding. The result is the once-monolithic task is now broken up into five separate tasks&mdash;one for each function.", width="800", height="211" %}
+  <figcaption>
+    The <code>saveSettings</code> function now executes its child functions as separate tasks.
+  </figcaption>
+</figure>
+
+The benefit of using a promise-based approach to yielding rather than manual use of `setTimeout` is better ergonomics. Yield points become declarative, and therefore easier to write, read, and understand.
+
+### Yield only when necessary
+
+What if you have a bunch of tasks, but you only want to yield if the user attempts to interact with the page? That's the kind of thing that [`isInputPending`](/isinputpending/) was made for!
+
+`isInputPending` is a function you can run at any time to determine if the user is attempting to interact with a page element: a call to `isInputPending` will return `true`. It returns `false` otherwise.
+
+Say you have a queue of tasks you need to run, but you don't want to get in the way of any inputs, such as an attempt to open a navigation menu, for example. This code which uses both `isInputPending` and our custom `yieldToMain()` function ensures that the input won't be delayed while the user is trying to interact with the page:
+
+```js
+async function saveSettings () {
+  // A task queue of functions
+  const tasks = [
+    validateForm,
+    showSpinner,
+    saveToDatabase,
+    updateUI,
+    sendAnalytics
+  ];
+  
+  while (tasks.length > 0) {
+    // Yield to a pending user input:
+    if (navigator.scheduling.isInputPending()) {
+      // There's a pending user input. Yield here:
+      await yieldToMain();
+    } else {
+      // Shift the the task out of the queue:
+      const task = tasks.shift();
+
+      // Run the task:
+      task();
+    }
+  }
+}
+```
+
+While `saveSettings` runs, it will loop over the tasks in the queue. If `isInputPending` ever returns `true` during the loop, `saveSettings` will call `yield` so the user input can be handled. Otherwise, it will shift the next task off the front of the queue and run it continuously. It will do this until no more tasks are left.
```

Comment:
```suggestion
While `saveSettings` runs, it will loop over the tasks in the queue. If `isInputPending()` ever returns `true` during the loop, `saveSettings` will call `yield` so the user input can be handled. Otherwise, it will shift the next task off the front of the queue and run it continuously. It will do this until no more tasks are left.
```